crates/ursa-network/src/service.rs

```diff
@@ -609,6 +612,11 @@ where
                             }
                         }
                         RequestType::StoreSummary(cache_summary) => {
+                            if self.peer_cached_content.len() == self.max_cache_summaries {
+                                let key_to_remove =
+                                    *self.peer_cached_content.iter().next().unwrap().0;
```

Comment:
It might be more readable.

```suggestion
                                let key_to_remove =
                                    *self.peer_cached_content.keys().next().unwrap();
```